### DISCUSSION TITLE
Allow setting the scope for the uniqueness rule in the storage tree

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5942] [Feature Requests]
> I want to set a rule for our storage tree that limits the names of containers in their parent of the parent. So, if we have an Aisle > Cabinets > Container, the user won’t be able to create two containers with the same name in an aisle.
> However, since the Storage tree is at the institution level, the uniqueness rules must apply globally, while this rule is only relevant for one division. Hence, I keep getting ‘duplicates exists’ warnings that prevent me from setting this rule.
> 
> It would be nice to have the option to set the scope of uniqueness of a rule only for a part of the tree, one branch and not the other (for example, all aisles within a specific room).

To implement this feature:
-  Introduce a scoped uniqueness setting for rules in the storage tree.
-  Allow users to specify a specific branch or part of the tree where uniqueness rules apply.
-  Users can define unique naming constraints for a particular division or section of the storage tree hierarchy.

**Requested by:** Eyal at The Hebrew University of Jerusalem

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/setting-scope-for-uniqueness-rule-in-storage-tree/1684/2
